### PR TITLE
Use backoff functionality from github.com/grafana/dskit

### PR DESCRIPTION
**Pull request by aknuds1**
**What this PR does**:
In this PR I propose to refactor to use [dskit](github.com/grafana/dskit)'s `backoff` package instead of pkg/util/backoff.go.

**Which issue(s) this PR fixes**:

**Checklist**
- [x] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 ## master / unreleased
 
+* [CHANGE] Use github.com/grafana/dskit/backoff instead of pkg/util/backoff.go.
```
Comment (pstibrany) [inline CHANGELOG.md:5]: I don't think we need changelog entry for this, as this isn't user-visible change.

**Review by pstibrany — APPROVED** [review]
Thank you!

File: CHANGELOG.md (hunk 1 above)
Comment (aknuds1) [reply CHANGELOG.md:5]: @pstibrany OK, I will remove it :)

**Review by treid314 — APPROVED** [review]
Nice! Thank you!

**Review by pracucci — APPROVED** [review]
LGTM, thanks! I've checked the updated dependencies and LGTM too.

**Review by pstibrany — APPROVED** [review]

**Comment by bboreham** [comment]
When other libraries are upgraded, e.g. `etcd/client`, it's good to call that out and check what changed.
(it looks mostly cosmetic)

**Pull request merged**